### PR TITLE
Build API Wrapper for Stripe, and write unit/feature tests for that

### PR DESCRIPTION
**Pull request by durrellchamorro**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -7,12 +7,23 @@ def new
...
+        description: "Registration charge for #{@user.email_address}"
+      )
+      if charge.successful?
+        @user.save
+        handle_invitation
+        AppMailer.delay.send_welcome_email(@user)
+        flash[:success] = "Thank you for registering with MyFlix. Please sign in."
+        redirect_to login_path
+      else
+        flash[:danger] = charge.error_message
```
Comment (tomtomecek) [inline app/controllers/users_controller.rb]: You should use `flash.now` when you render.

HOMEWORK: find out why it is better option to use `now` while rendering template.

File: spec/features/user_invites_a_friend_spec.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require "spec_helper"
 
 feature "User successfully invites a friend" do
-  scenario "user invites a friend and the friend signs up", js: true do
+  scenario "user invites a friend and the friend signs up", js: true, vcr: true do
```
Comment (tomtomecek) [inline spec/features/user_invites_a_friend_spec.rb]: I prefer to use symbols directly and not hash. `:js, :vcr` its more succinct.

File: spec/features/user_registers_spec.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  scenario "with valid user info and invalid card", js: true, vcr: true  do
+    fill_in_valid_user_info
+    fill_in_invalid_card_info
+    click_button("Sign Up")
+
+    expect(User.count).to eq(0)
+    expect(page).to have_content("This card number looks invalid.")
+  end
+
+  scenario "with valid user info and declined card", js: true, vcr: true  do
```
Comment (tomtomecek) [inline spec/features/user_registers_spec.rb]: If all specs use `, js: true, vcr: true` maybe you could refactor it to the `feature` level.

File: spec/features/user_registers_spec.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  def fill_in_invalid_user_info
+    fill_in("Email Address", with: "example@gmail.com")
+    fill_in("Password", with: "p")
+    fill_in("Full Name", with: "ralph")
+  end
+
+  def fill_in_declined_card_info
+    fill_in "CreditCardNumber", with: '4000000000000002'
+    fill_in("SecurityCode", with: '123')
+    select('2019', :from => 'date_year')
```
Comment (tomtomecek) [inline spec/features/user_registers_spec.rb]: You should not mix the style of hashes. `from: 'date_year'` vs `:from => 'date_year'`  the latter looks weird when everything else is new Ruby hash styled.

File: spec/models/stripe_wrapper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            )
+          expect(response).to be_successful
+        end
+      end
+
+      context "with invalid card" do
+        let(:token) { set_stripe_token_for_card("4000000000000002") }
+        it "makes a card declined charge" do
+          response = StripeWrapper::Charge.create(
+            :amount => 999,
```
Comment (tomtomecek) [inline spec/models/stripe_wrapper_spec.rb]: the indentation level for `:amount => 999` is wrong. Keys should be aligned.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (durrellchamorro) [inline app/controllers/users_controller.rb]: Oops. The flash will linger to the page that comes after the rendered page if I don't do flash.now.

**Pull request merged**